scarb/src/ops/metadata.rs

```diff
@@ -270,6 +262,50 @@ fn collect_compilation_unit_metadata(
         .unwrap()
 }
 
+fn collect_proc_macro_compilation_unit_metadata(
+    compilation_unit: &ProcMacroCompilationUnit,
+) -> m::CompilationUnitMetadata {
+    let components = collect_compilation_unit_components(compilation_unit.components.iter());
+    m::CompilationUnitMetadataBuilder::default()
+        .id(compilation_unit.id())
+        .package(wrap_package_id(compilation_unit.main_package_id))
```

Comment:
If the method `main_package_id` stays, shouldn't it be used here? A bit of an overhead, but we should stay consistent with it

Reply:
Note, that the whole existence of the `main_package_id()` is rather a technicality - so it can be used in default trait impls. All the fields in both CU structs are still public, so can be used outside it's implementation without getters.